kythe/java/com/google/devtools/kythe/analyzers/base/AbstractCompilationAnalyzer.java

```diff
@@ -65,6 +66,8 @@ public void analyzeRequest(AnalysisRequest req, FactEmitter emitter) throws Anal
           "Analyzing compilation: {%s}",
           TextFormat.shortDebugString(req.getCompilation().getVName()));
       analyzeCompilation(req.getCompilation(), Optional.ofNullable(revision), fileData, emitter);
+    } catch (InterruptedException e) {
```

Comment:
Thread.currentThread().interrupt();

Reply:
Based on the last paragraph on this page, this should not reset the interrupt bit since, by convention, the interrupt bit should be cleared when throwing an `InterruptedException`: https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html